app/client/jsx/Room.jsx

```diff
@@ -39,17 +42,41 @@ class Room extends Component {
             iframe: true
         }
 
+        this.state = {
+            room,
+            entered: false,
+            users: []
+        }
+
+        this.httpApi = new HttpApi()
+    }
+
+    async fetchUsersForRoom(room) {
+        const { success, users } = await this.httpApi.getUsers(room || this.state.room)
```

Comment:
Ideally I'd merge my PR first, and then you'd change this to a `httpApi.getUsers()` call for all of the users. We only fetch the whole state once now and no longer make requests for users per room.